app/scripts/modules/core/src/managed/EnvironmentsList.tsx

```diff
@@ -48,11 +49,14 @@ export function EnvironmentsList({
                     application={application}
                     key={resource.id}
                     resource={resource}
-                    environment={name}
-                    showReferenceName={allArtifacts.length > 1}
-                    artifactVersionsByState={artifactVersionsByState}
-                    artifactDetails={artifactDetails}
-                    depth={0}
+                    metadata={
+                      <ResourceDeploymentStatus
+                        environment={name}
+                        showReferenceName={allArtifacts.length > 1}
+                        artifactVersionsByState={artifactVersionsByState}
+                        artifactDetails={artifactDetails}
+                      />
```

Comment:
Here, I would prefer to either:

- keep the props as they were before, and render `ResourceDeploymentStatus`  within the `ManagedResourceObject` file.
- or just directly render `ResrouceDeploymentStatus` as a child of the component
```js
<ManagedResourceObject {...otherProps}>
  <ResourceDeploymentStatus
    environment={name}
    showReferenceName={allArtifacts.length > 1}
    artifactVersionsByState={artifactVersionsByState}
    artifactDetails={artifactDetails}
  />
</ManagedResourceObject>
```

Reply:
> keep the props as they were before
I rather not to, because they are only needed in one case.

> or just directly render ResrouceDeploymentStatus as a child of the component
What's the difference between assigning it to `metadata` and `children`? I agree that your implementation looks nice, but I suspect that it could be confusing (to use children for the metadata)

Reply:
I think it’s clearer to specify a JSX element’s children using the native `props.children` that JSX has.  The `ResourceDeploymentStatus` component is really just a child component of `ManagedResourceObject`.